core/src/test/java/org/apache/iceberg/TestMetadataTableScans.java

```diff
@@ -159,6 +162,105 @@ public void testAllManifestsTableHonorsIgnoreResiduals() throws IOException {
+    validateIncludes(tasksNoFilter, 2);
+    validateIncludes(tasksNoFilter, 3);
+
+    Expression andEquals = Expressions.and(
+        Expressions.equal("partition.data_bucket", 0),
+        Expressions.greaterThan("record_count", 0));
+    TableScan scanAndEq = partitionsTable.newScan()
+        .filter(andEquals);
+    CloseableIterable<FileScanTask> tasksAndEq = ((PartitionsTable) partitionsTable)
+        .planTasks((StaticTableScan) scanAndEq);
+    Assert.assertEquals(1, Iterators.size(tasksAndEq.iterator()));
+    validateIncludes(tasksAndEq, 0);
+
+    Expression ltAnd = Expressions.and(
+        Expressions.lessThan("partition.data_bucket", 2),
+        Expressions.greaterThan("record_count", 0));
+    TableScan scanLtAnd = partitionsTable.newScan()
+        .filter(ltAnd);
+    CloseableIterable<FileScanTask> tasksLtAnd = ((PartitionsTable) partitionsTable)
+        .planTasks((StaticTableScan) scanLtAnd);
+    Assert.assertEquals(2, Iterators.size(tasksLtAnd.iterator()));
+    validateIncludes(tasksLtAnd, 0);
+    validateIncludes(tasksLtAnd, 1);
+
+    Expression or = Expressions.or(
+        Expressions.equal("partition.data_bucket", 2),
+        Expressions.greaterThan("record_count", 0));
+    TableScan scanOr = partitionsTable.newScan()
+        .filter(or);
+    CloseableIterable<FileScanTask> tasksOr = ((PartitionsTable) partitionsTable)
+        .planTasks((StaticTableScan) scanOr);
+    Assert.assertEquals(4, Iterators.size(tasksOr.iterator()));
+    validateIncludes(tasksOr, 0);
+    validateIncludes(tasksOr, 1);
+    validateIncludes(tasksOr, 2);
+    validateIncludes(tasksOr, 3);
+
+    Expression not = Expressions.not(
+        Expressions.lessThan("partition.data_bucket", 2));
+    TableScan scanNot = partitionsTable.newScan()
+        .filter(not);
+    CloseableIterable<FileScanTask> tasksNot = ((PartitionsTable) partitionsTable)
+        .planTasks((StaticTableScan) scanNot);
+    Assert.assertEquals(2, Iterators.size(tasksNot.iterator()));
+    validateIncludes(tasksNot, 2);
+    validateIncludes(tasksNot, 3);
+
+    Expression set = Expressions.in("partition.data_bucket", 2, 3);
+    TableScan scanSet = partitionsTable.newScan()
+        .filter(set);
+    CloseableIterable<FileScanTask> tasksSet = ((PartitionsTable) partitionsTable)
+        .planTasks((StaticTableScan) scanSet);
+    Assert.assertEquals(2, Iterators.size(tasksSet.iterator()));
+    validateIncludes(tasksSet, 2);
+    validateIncludes(tasksSet, 3);
+
+    Expression unary = Expressions.notNull("partition.data_bucket");
+    TableScan scanUnary = partitionsTable.newScan()
+        .filter(unary);
+    CloseableIterable<FileScanTask> tasksUnary = ((PartitionsTable) partitionsTable)
+        .planTasks((StaticTableScan) scanUnary);
+    Assert.assertEquals(4, Iterators.size(tasksUnary.iterator()));
+    validateIncludes(tasksUnary, 0);
+    validateIncludes(tasksUnary, 1);
+    validateIncludes(tasksUnary, 2);
+    validateIncludes(tasksUnary, 3);
```

Comment:
I haven't looked very closely, but this appears to be a really long test. My guess is that you're mixing test cases that should be in separate methods. Can you clean it up so that you're testing cases individually?

Reply:
Yes, done